dsl/src/commonMain/kotlin/it/unibo/collektive/aggregate/api/Aggregate.kt

```diff
@@ -0,0 +1,65 @@
+package it.unibo.collektive.aggregate.api
+
+import it.unibo.collektive.ID
+import it.unibo.collektive.aggregate.api.YieldingContext.YieldingResult
+import it.unibo.collektive.field.Field
+
+typealias YieldingScope<Initial, Return> = YieldingContext<Initial, Return>.(Initial) -> YieldingResult<Initial, Return>
+
+/**
+ * Models the minimal set of aggregate operations.
+ * Holds the [localId] of the device executing the aggregate program.
+ */
+interface Aggregate {
+    /**
+     * The local [ID] of the device.
+     */
+    val localId: ID
+
+    /**
+     * This function computes the local value of e_i, substituting variable n with the nvalue w of
+     * messages received from neighbours, using the local value of e_i ([initial]) as a default.
+     * The exchange returns the neighboring or local value v_r from the evaluation of e_r applied to the [body].
+     * e_s evaluates to a nvalue w_s consisting of local values to be sent to neighbour devices δ′,
+     * which will use their corresponding w_s(δ') as soon as they wake up and perform their next execution round.
+     *
+     * Often, expressions e_r and e_s coincide, so this function provides a shorthand for exchange(e_i, (n) => (e, e)).
+     *
+     * ## Example
+     * ```
+     * exchange(0) { f ->
+     *  f.mapField { _, v -> if (v % 2 == 0) v + 1 else v * 2 }
+     * }
+     * ```
+     * The result of the exchange function is a field with as messages a map with key the id of devices across the
+     * network and the result of the computation passed as relative local values.
+     */
+    fun <Initial> exchange(initial: Initial, body: (Field<Initial>) -> Field<Initial>): Field<Initial>
+
+    /**
+     * Same behavior of [exchange] but this function can yield a [Field] of [Return] value.
+     */
+    fun <Initial, Return> exchanging(
+        initial: Initial,
+        body: YieldingScope<Field<Initial>, Field<Return>>,
+    ): Field<Return>
```

Comment:
It has been suggested to add a boolean parameter `strictSemantic` to `exchange` and `exchanging` to allow also the use of its implementation as proposed in the [paper](https://doi.org/10.4230/LIPIcs.ECOOP.2022.20) 
Wouldn't be better if we added it directly to the class `AggregateContext`?

Reply:
Why should we add it in the `AggregateContext` that is the internal implementation of the `Aggregate` interface? We should put the flag in the interface definition if we want to support both behaviors.
I am missing something?

Reply:
@DanySK @angelacorte What do you think if we open an issue about this comment and merge this PR since it is becoming very big?

Reply:
makes sense to me